src/Security/Authenticator/ApiKeyAuthenticator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Security\Authenticator;
+
+use App\Security\Model\ApiTokenUser;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\AbstractGuardAuthenticator;
+
+class ApiKeyAuthenticator extends AbstractGuardAuthenticator
+{
+    protected string $apiKeyHeaderName;
+    protected string $apiAllowedKeys;
+
+    public function __construct(string $apiKeyHeaderName, string $apiAllowedKeys)
+    {
+        $this->apiKeyHeaderName = $apiKeyHeaderName;
+        $this->apiAllowedKeys = $apiAllowedKeys;
+    }
+
+    public function start(Request $request, AuthenticationException $authException = null): Response
+    {
+        return new Response(null, 401);
+    }
+
+    public function supports(Request $request): bool
+    {
+        return true;
+    }
+
+    public function getCredentials(Request $request): string
+    {
+        return $request->headers->get($this->apiKeyHeaderName, '');
+    }
+
+    public function getUser($credentials, UserProviderInterface $userProvider): ?UserInterface
+    {
+        return new ApiTokenUser($credentials);
+    }
+
+    public function checkCredentials($credentials, UserInterface $user): bool
+    {
+        $allowedTokens = explode(',', $this->apiAllowedKeys);
+        $allowedTokens = array_filter($allowedTokens, static fn (string $allowedToken) => !empty($allowedToken));
+        $allowedTokens = array_map('trim', $allowedTokens);
```

Comment:
Doing `trim` before filtering out empty tokens will disallow empty tokens as valid tokens in case someone puts few spaces in the token, eg `TOKENS=token123,  ,token1234`
```suggestion
        $allowedTokens = array_map('trim', $allowedTokens);
        $allowedTokens = array_filter($allowedTokens, static fn (string $allowedToken) => !empty($allowedToken));
```